magicctapipe/scripts/lst1_magic/semi_automatic_scripts/DL1_to_DL2.py

```diff
@@ -0,0 +1,302 @@
+        File_list = glob.glob(f"{outdir}/ST*.txt")
+        night_date = night.split("/")[-1]
+        if str(night_date) not in LST_date:
+            continue
+
+        for file in File_list:
+            with open(file, "r") as f:
+                process_size = len(f.readlines()) - 1
+            if process_size < 0:
+                continue
+            nsb = file.split("/")[-1].split("_")[1]
+            period = file.split("/")[-1].split("_")[0]
+            dec = df_LST[df_LST.source == source].iloc[0]["MC_dec"]
+            if np.isnan(dec):
+                print(f"MC_dec is NaN for {source}")
+                continue
+            dec = str(dec).replace(".", "")
+            dec = str(dec).replace("-", "min_")
+            RFdir = f"{RF_dir}/{period}/NSB{nsb}/v{MC_v}/dec_{dec}/"
+            if (not os.path.isdir(RFdir)) or (len(os.listdir(RFdir)) == 0):
+                print(f"no RF availables in {RFdir}")
+                continue
+            slurm = slurm_lines(
+                queue="short",
+                job_name=f"{process_name}_DL1_to_DL2",
+                nice_parameter=nice,
+                array=process_size,
+                mem="150g",
+                out_name=f"{outdir}/slurm-%x.%A_%a",
+            )
+            rc = rc_lines(
+                store="$SAMPLE ${SLURM_ARRAY_JOB_ID} ${SLURM_ARRAY_TASK_ID}",
+                out=f"{outdir}/list_{nsb}_{period}",
+            )
+            out_file = outdir.rstrip("/logs")
+            lines = (
+                slurm
+                + [
+                    f"SAMPLE_LIST=($(<{file}))\n",
+                    "SAMPLE=${SAMPLE_LIST[${SLURM_ARRAY_TASK_ID}]}\n",
+                    f"export LOG={outdir}",
+                    "/DL1_to_DL2_${SLURM_ARRAY_JOB_ID}_${SLURM_ARRAY_TASK_ID}.log\n",
+                    f"conda run -n {env_name} lst1_magic_dl1_stereo_to_dl2 --input-file-dl1 $SAMPLE --input-dir-rfs {RFdir} --output-dir {out_file} >$LOG 2>&1\n\n",
+                ]
+                + rc
+            )
+            with open(
+                f'{source}_DL1_to_DL2_{file.split("/")[-1].rstrip("txt")}sh',
+                "w",
+            ) as f:
+                f.writelines(lines)
+
+
+def main():
+    """
+    Here we read the config_auto_MCP.yaml file and call the functions defined above.
+    """
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--config-file",
+        "-c",
+        dest="config_file",
+        type=str,
+        default="./config_auto_MCP.yaml",
+        help="Path to a configuration file",
+    )
+
+    args = parser.parse_args()
+    with open(
+        args.config_file, "rb"
+    ) as f:  # "rb" mode opens the file in binary format for reading
+        config = yaml.safe_load(f)
+
+    target_dir = Path(config["directories"]["workspace_dir"])
+    RF_dir = config["directories"]["RF"]
+    env_name = config["general"]["env_name"]
+    ST_list = config["needed_parameters"]["ST_list"]
```

Comment:
it would be more convenient to have the ST periods as a dictionary, the key would be ST period name, and value would be a two element list of validity dates